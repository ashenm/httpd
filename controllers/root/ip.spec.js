const ip = require('./ip');

describe('root/ip', () => {
  const response = {
    json: jest.fn().mockReturnThis(),
    status: jest.fn().mockReturnThis()
  };

  // when
  beforeAll(() => {
    ip({ ip: 'test-ip-address' }, response);
  });

  // then
  it('Should dispatch response status with valid arity', () => {
    expect(response.status.mock.calls).toMatchSnapshot();
  });

  it('Should dispatch response body with valid arity', () => {
    expect(response.json.mock.calls).toMatchSnapshot();
  });

  afterAll(jest.clearAllMocks);
});
