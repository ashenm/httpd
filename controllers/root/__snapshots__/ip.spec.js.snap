// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root/ip Should dispatch response body with valid arity 1`] = `
Array [
  Array [
    Object {
      "ip": "test-ip-address",
    },
  ],
]
`;

exports[`root/ip Should dispatch response status with valid arity 1`] = `
Array [
  Array [
    200,
  ],
]
`;
